### ISSUE TITLE
hamming distance

### ISSUE THREAD

**Issue by 6Youcai** [OpenGene/OpenGene.jl#9]
In the function hamming_distance(s1,s2), len = min(length(s1), length(s2)),  while Hamming distance are between two strings of equal length, see https://en.wikipedia.org/wiki/Hamming_distance

**Comment by sfchen** [comment]
Yes, this hamming_distance is generalized for comparing sequences with different length
